config/src/main/scala/io/buoyant/config/Config.scala

```diff
@@ -0,0 +1,9 @@
+package io.buoyant.config
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonTypeInfo}
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "kind", visible = true)
```

Comment:
Do all Config types actually have a kind?  clients and servers, for instance, do not.  Perhaps call this PolymorphicConfig or something that indicates when it needs to be used?  It should have a comment that clarifies what it's for, what needs to extend it, etc.